rfcs/convergence/array-slots.md

```diff
@@ -0,0 +1,165 @@
+RFC: Array slots
+
+---
+
+@smhigley
+
+## Summary
+
+This RFC proposes modifying the Slots API to accept arrays of slot props to support components like Combobox or List, where the most ergonomic authoring experience is passing in an array of data rather than unique JSX children.
+
+## Problem Statement
+
+So far, all slot usage in vNext components assumes the desired authoring experience is to define slots or children one at a time. That makes sense for the components we've created so far, since none of them render a set of elements that are usually generated from an array of data. Even for Menu, which contains a list of MenuItem children, it's not common for authors to want to convert an array of repetitive data into menu items.
+
+In contrast, Combobox and Listbox patterns do commonly take an array of data and use it to render mostly uniform items. The same is likely going to be true of components like List (if we have one) and DetailsList/Grid. For those controls, good authoring ergonomics lean more towards this:
+
+```js
+<Combobox options={['apple', 'banana', 'cucumber', ...etc]} />
+```
+
+Than this:
+
+```js
+<Combobox>
```

Comment:
The `map` does not need to event be rendered directly as children

```jsx
const options = ["apple", "banana", "cucumber"];

function App() {
  const comboboxItems = options.map(item => (<Option key={item.id}>{item.name}</Option>))
  return (
    <>
      <Combobox>
        {comboboxItems}
      </Combobox>
    </>
  );
}
```

Reply:
I prefer this approach to having two props of `items` and `onRenderItems` which is what we did in v8. It's effectively the same thing, but the mapping is done by the user instead of the component.